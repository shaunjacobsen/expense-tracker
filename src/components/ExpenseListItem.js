import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = (props) => (
  <div>
    <strong>${props.amount / 100}</strong> <Link to={`/edit/${props.id}`}>{props.description}</Link>
  </div>
);

export default ExpenseListItem;