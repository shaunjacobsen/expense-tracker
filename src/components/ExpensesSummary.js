import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from './../selectors/expenses';
import expensesTotal from './../selectors/expenses-total';

export const ExpensesSummary = props => (
  <div>
    Viewing {props.expenseCount} expense{props.expenseCount === 1
      ? ' '
      : 's '}
    totalling {numeral(props.total / 100).format('$0,0.00')}
  </div>
);

const mapStateToProps = state => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    total: expensesTotal(visibleExpenses),
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
