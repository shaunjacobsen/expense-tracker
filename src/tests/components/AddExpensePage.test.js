import React from 'react';
import { shallow } from 'enzyme';
import { AddExpensePage } from './../../components/AddExpensePage';
import expenses from './../fixtures/expenses';

let onSubmit, history, wrapper;

beforeEach(() => {
  onSubmit = jest.fn();
  history = { push: jest.fn() };
  wrapper = shallow(<AddExpensePage addExpense={onSubmit} history={history} />);
});

test('it should render AddExpensePage correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('it should handle onSubmit', () => {
  wrapper.find('ExpenseForm').prop('onSubmit')(expenses[1]);
  expect(history.push).toHaveBeenLastCalledWith('/');
  expect(onSubmit).toHaveBeenLastCalledWith(expenses[1]);
});