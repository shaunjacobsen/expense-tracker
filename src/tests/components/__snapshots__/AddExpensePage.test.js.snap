// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render AddExpensePage correctly 1`] = `
<div>
  <h2>
    Add Expense
  </h2>
  <ExpenseForm
    onSubmit={[Function]}
  />
</div>
`;
